opds_import.py

```diff
@@ -65,10 +65,14 @@ def parse_identifier(db, identifier):
     :rtype: Optional[core.model.identifier.Identifier]
     """
     parsed_identifier = None
-    result = Identifier.parse_urn(db, identifier)
 
-    if result is not None:
-        parsed_identifier, _ = result
+    try:
+        result = Identifier.parse_urn(db, identifier)
```

Comment:
Are these changes related to the DPLA download issue? When were we seeing an exception parsing an identifier?

Reply:
Yes, they're related. When I was testing my changes I saw at least one DPLA identifier that wasn't a valid ISBN even though it claimed to be (by using `urn:ISBN` scheme).